packages/core/src/shipping/strategies/stripe-upe/stripe-upe-shipping-strategy.ts

```diff
@@ -0,0 +1,118 @@
+        const { stripeupe: { container, gatewayId, methodId, onChangeShipping, getStyles, availableCountries } = {} } = options;
+
+        if (!methodId || !gatewayId) {
+            throw new InvalidArgumentError('Unable to proceed because "methodId " or "gatewayId" argument is not provided.');
+        }
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(gatewayId, { params: { method: methodId } }));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId, gatewayId);
+        const { initializationData: { stripePublishableKey, stripeConnectedAccount } } = paymentMethod;
+
+        if (!paymentMethod || !paymentMethod.initializationData.stripePublishableKey || !paymentMethod.clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._stripeUPEClient = await this._stripeUPEScriptLoader.getStripeClient(stripePublishableKey, stripeConnectedAccount);
+
+        let appearance: StripeUPEAppearanceOptions;
+        const styles = getStyles && getStyles();
+
+        if (styles) {
+            appearance = {
+                variables: {
+                    colorPrimary: styles.fieldInnerShadow,
+                    colorBackground: styles.fieldBackground,
+                    colorText: styles.labelText,
+                    colorDanger: styles.fieldErrorText,
+                    colorTextSecondary: styles.labelText,
+                    colorTextPlaceholder: styles.fieldPlaceholderText,
+                    spacingUnit: '4px',
+                    borderRadius: '4px',
+                },
+                rules: {
+                    '.Input': {
+                        borderColor: styles.fieldBorder,
+                        color: styles.fieldText,
+                        boxShadow: styles.fieldInnerShadow,
+                    },
+                }
+            };
+        } else {
+            appearance = {
+                variables: {
+                    spacingUnit: '4px',
+                    borderRadius: '4px',
+                }
+            };
+        }
+
+        this._stripeElements = this._stripeUPEScriptLoader.getElements(this._stripeUPEClient, {
+                clientSecret: paymentMethod.clientToken,
+                appearance: appearance,
+            });
+
+        const shippingAddressElement = this._stripeElements.getElement(StripeElementType.SHIPPING) ||
+            this._stripeElements.create(StripeElementType.SHIPPING, {allowedCountries: [availableCountries]});
+
+        shippingAddressElement.on('change', (event: StripeEventType) => {
+            if (!('isNewAddress' in event)) {
+                throw new MissingDataError(MissingDataErrorType.MissingCustomer);
+            } else {
+                if ((event.complete || event.isNewAddress) && onChangeShipping && typeof onChangeShipping === 'function') {
```

Comment:
By the interface definition it seems like `onChangeShipping` is a required value which is a function? In that case do we need to check for it?